hotsos/core/host_helpers/cli.py

```diff
@@ -28,20 +29,26 @@
 from hotsos.core.log import log
 
 
-class CmdBase():
-    """ Base class for all command source types. """
-    def __init__(self):
+@dataclass
+class CmdBase:
+    """ Base class for all command source types.
+
+    Provides a way to save original state and restore to that state.
+    """
+    def __post_init__(self):
         self.hooks = {}
-        self.reset()
+        # make a copy of field original values.
+        for f in fields(self):
+            setattr(self, 'original_' + f.name, f.type(getattr(self, f.name)))
+        super().__init__()
```

Comment:
This is to support the case where CmdBase or an implementation thereof is used in multiple inheritance where one or more class has a __init__ that needs to be called. Currently I have structure things so that this is not necessarily needed but it is good practice to leave it so that in the future if any class used in conjunction with this one did implement __init__ it would get called - although because this is a dataclass it would not be able to pass any args/kwargs through to the parent.

Reply:
ive gone ahead and removed that line. I agree that have it there given that it has no actual use in the current code is not right.